src/control/lib/control/pool.go

```diff
@@ -423,17 +423,21 @@ type (
 	// PoolQueryTargetReq contains parameters for a pool query target request
 	PoolQueryTargetReq struct {
 		poolRequest
-		ID        string
-		Rank      system.Rank
-		Targetidx []uint32
+		ID      string
+		Rank    system.Rank
+		Targets []uint32
 	}
 
+	StorageMediaType int32
+
 	// StorageTargetUsage represents DAOS target storage usage
 	StorageTargetUsage struct {
-		Total uint64 `json:"total"`
-		Free  uint64 `json:"free"`
+		Total     uint64           `json:"total"`
+		Free      uint64           `json:"free"`
+		MediaType StorageMediaType `json:"media_type"`
 	}
 
+	// TargetPerf - placeholder. TODO: storage/network bandwidth, latency, etc.
 	TargetPerf struct {
 		Foo int32 `json:"foo"`
 	}
```

Comment:
I see that you removed this from the protobuf messages. What is the benefit of leaving it here? If there is a plan to iterate relatively quickly and actually use this, then I'm fine with landing as-is, but if this is potentially going to sit around indefinitely then I think it'd be better to remove it as it's a bit of a wart.

Reply:
OK let me see if I can remove it also from here. The C.daos_pool_query() info call fills in a daos_target_info_t containing this. So I just took what it filled in its structure and converted it into this Go data type. I don't know maybe I assumed it would be needed locally after removing from protobuf. Did lots of iteration on the patch trying to get the (hacky) protobuf to control type conversion stuff actually working and didn't focus too much on this placeholder element since there were some other issues in that area to solve.

Reply:
no near-term plans on my part to populate that placeholder content.